src/main/scala/com/campudus/tableaux/database/DatabaseConnection.scala

```diff
@@ -19,40 +19,51 @@ class DatabaseConnection(verticle: Verticle) {
 
   case class Transaction(msg: Message[JsonObject]) {
 
-    def query(query: String, values: JsonArray): Future[(Transaction, JsonObject)] = transactionHelper(Json.obj(
+    def query(query: String, values: JsonArray): Future[(Transaction, JsonArray)] = transactionHelper(Json.obj(
       "action" -> "prepared",
       "statement" -> query,
-      "values" -> values)) map { r => (Transaction(r), r.body()) }
+      "values" -> values)) flatMap { r => try Future(Transaction(r), resultChecker(r.body())) catch recover() }

```

Comment:
I know I had a recoverWith before this, but dont know why its a try / catch now. But I will fix this, also the name from the recover method